File: src/main/java/org/shark/example/service/base/async/AsyncTask1.java
package org.shark.example.service.base.async;

import lombok.extern.slf4j.Slf4j;
import org.springframework.scheduling.annotation.Async;
import org.springframework.scheduling.annotation.AsyncResult;
import org.springframework.stereotype.Service;

import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.concurrent.CompletableFuture;
import java.util.concurrent.Future;

@Service
@Slf4j
public class AsyncTask1 {

    @Async("taskExecutor1")
    public CompletableFuture<String> start(String value) {
        String time = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss").format(new Date());
//        log.info("time = " + time + ", value = " + value);
        try {
            Thread.sleep(3000);
        } catch (InterruptedException e) {
            throw new RuntimeException(e);
        }
        return CompletableFuture.completedFuture("AsyncTask1 value = " + value);
    }
}

File: src/main/kotlin/org/shark/example/service/base/async/AsyncTask1.kt
package org.shark.example.service.base.async

import org.slf4j.LoggerFactory
import org.springframework.scheduling.annotation.Async
import org.springframework.stereotype.Service
import java.text.SimpleDateFormat
import java.util.*
import java.util.concurrent.CompletableFuture

@Service
class AsyncTask1 {

    private val log = LoggerFactory.getLogger(AsyncTask1::class.java)

    @Async("taskExecutor1")
    fun start(value: String): CompletableFuture<String> {
        val time = SimpleDateFormat("yyyy-MM-dd HH:mm:ss").format(Date())
        // log.info("time = $time, value = $value")
        try {
            Thread.sleep(3000)
        } catch (e: InterruptedException) {
            throw RuntimeException(e)
        }
        return CompletableFuture.completedFuture("AsyncTask1 value = $value")
    }
}
